virttest/qemu_devices/qcontainer.py

```diff
@@ -2896,6 +2896,21 @@ def throttle_group_define_by_params(self, group_params, name):
         props = json.loads(group_params.get("throttle_group_parameters", "{}"))
         return QThrottleGroup(name, props)
 
+    def thread_context_define_by_params(self, params, name):
+        """
+        Create thread-context object from params.
+        """
+        tc_params = Params()
+        suffix = '_thread_context'
+        for key in list(params.keys()):
+            if key.endswith(suffix):
+                new_key = key.rsplit(suffix)[0]
+                tc_params[new_key] = params[key]
+        backend = "thread-context"
+        dev = qdevices.ThreadContext(backend, params=tc_params)
```

Comment:
Just wonder about if using `qdevices.QObject` to instantiate the objects would be fairly enough, or is there some reason of introducing this new child class?

Reply:
Mmm, probably using QObject is enough, thread-context object is quite simple
```bash
$ /usr/libexec/qemu-kvm -object thread-context,?
thread-context options:
  cpu-affinity=<int>
  node-affinity=<int>
```

Reply:
@YongxueHong what do you think about the idea of instantiate the object under QObject directly?

Reply:
Hi @mcasquer
It's okay to me for using the QObject directly.
Thanks.

Reply:
@luckyh @YongxueHong updated now!